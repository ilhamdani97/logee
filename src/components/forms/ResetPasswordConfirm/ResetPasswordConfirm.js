import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import styles from './styles.scoped.css';
import { Button } from 'logeeport-ui';
import { useSelector } from 'react-redux';
import Password from '@components/fields/Password';

const ResetPassword = (props) => {
  const { handleSubmit, valid } = props;

  const passwordInputs = { placeholder: 'Masukan kata sandi baru' };
  const repasswordInputs = { placeholder: 'Masukan ulang kata sandi baru' };

  const { message, isLoading } = useSelector((s) => s.resetPassword);
  return (
    <>
      <h3>
        Ubah Kata Sandi
      </h3>
      <form className={styles.root}>
        <Field
          className={styles.passwordSpacing}
          component={Password}
          inputProps={passwordInputs}
          label="Kata sandi baru"
          name="newPassword"
        />
        <Field
          component={Password}
          inputProps={repasswordInputs}
          label="Ulangi kata sandi baru"
          name="confirmPassword"
        />
        <Button
          disabled={isLoading || !valid}
          isLoading={isLoading}
          onClick={handleSubmit}
        >
          Ubah Kata Sandi
        </Button>
      </form>
      {message && <span className={styles.error}>{message}</span>}
    </>
  );
};

ResetPassword.defaultProps = {
  handleSubmit: () => {},
  valid: true,
};

ResetPassword.propTypes = {
  handleSubmit: PropTypes.func,
  valid: PropTypes.bool,
};

export default ResetPassword;
