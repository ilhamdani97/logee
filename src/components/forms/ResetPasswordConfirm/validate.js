export default function validate(values) {
  const { newPassword, confirmPassword } = values;

  return {
    newPassword: !newPassword && 'required' || newPassword.length < 6 && 'minimal 6 karakter',
    confirmPassword: !confirmPassword && 'required' || confirmPassword !== newPassword && 'tidak sesuai' || confirmPassword.length < 6 && 'minimal 6 karakter',
  };
}
