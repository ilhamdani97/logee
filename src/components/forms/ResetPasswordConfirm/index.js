import { reduxForm } from 'redux-form';
import Component from './ResetPasswordConfirm';
import validate from './validate';

export default reduxForm({
  form: 'resetpasswordconfirm',
  initialValues: {
    newPassword:'',
    confirmPassword:''
  },
  validate,
})(Component);
