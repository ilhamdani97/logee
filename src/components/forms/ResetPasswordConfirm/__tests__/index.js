import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ResetPasswordConfirm from '../ResetPasswordConfirm';
import { useSelector } from 'react-redux';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    isLoading:false,
    message: 'test'
  };
});

describe('src/components/forms/ResetPasswordConfirm', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ResetPasswordConfirm />);
    expect(tree).toMatchSnapshot();
  });
});
