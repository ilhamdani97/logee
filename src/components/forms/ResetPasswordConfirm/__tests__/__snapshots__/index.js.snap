// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ResetPasswordConfirm render 1`] = `
<React.Fragment>
  <h3>
    Ubah Kata Sandi
  </h3>
  <form>
    <Field
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": [Function],
          "type": [Function],
        }
      }
      inputProps={
        Object {
          "placeholder": "Masukan kata sandi baru",
        }
      }
      label="Kata sandi baru"
      name="newPassword"
    />
    <Field
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": [Function],
          "type": [Function],
        }
      }
      inputProps={
        Object {
          "placeholder": "Masukan ulang kata sandi baru",
        }
      }
      label="Ulangi kata sandi baru"
      name="confirmPassword"
    />
    <Button
      buttonProps={Object {}}
      className=""
      color="default"
      disabled={false}
      fullwidth={false}
      isLoading={false}
      loadingType="text"
      onClick={[Function]}
      size="medium"
    >
      Ubah Kata Sandi
    </Button>
  </form>
  <span>
    test
  </span>
</React.Fragment>
`;
